drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/referral_postings_loader.rb

```diff
@@ -0,0 +1,323 @@
+          enrollment.attributes = {
+            personal_id: personal_id,
+            entry_date: entry_date,
+            household_id: household_id,
+            relationship_to_hoh: relationship_to_hoh(member_row),
+            enrollment_coc: enrollment_coc_by_project_id[project_id],
+          }
+          enrollment.build_wip(
+            date: entry_date,
+            client_id: client_pk,
+            project_id: project_pk_by_id(project_id),
+          )
+          enrollment
+        end.compact
+      end.compact
+    end
+
+    def build_referral_records
+      seen = Set.new
+      posting_rows.map do |row|
+        referral_id = row_value(row, field: 'REFERRAL_ID')
+        next if seen.include?(referral_id)
+
+        seen.add(referral_id)
+        found_household_member = household_member_rows_by_referral(referral_id).detect do |member_row|
+          mci_id = row_value(member_row, field: 'MCI_ID')
+          client_pk_by_mci_id(mci_id)
+        end
+        if found_household_member.nil?
+          log_info "#{row.context} skipping referral ID \"#{referral_id}\" - could not resolve any household member MCI IDs"
+          next
+        end
+
+        referral_class.new(
+          # NOTE: since the referral comes from link, the enrollment id should be NULL
+          identifier: referral_id,
+          referral_date: parse_date(row_value(row, field: 'REFERRAL_DATE')),
+          service_coordinator: row_value(row, field: 'SERVICE_COORDINATOR'),
+          referral_notes: row_value(row, field: 'REFERRAL_NOTES', required: false),
+          chronic: yn_boolean(row_value(row, field: 'CHRONIC', required: false)),
+          score: row_value(row, field: 'SCORE', required: false),
+          needs_wheelchair_accessible_unit: yn_boolean(row_value(row, field: 'NEEDS_WHEELCHAIR_ACCESSIBLE_UNIT', required: false)),
+        )
+      end
+    end
+
+    # assign inferred unit occupancy
+    def assign_unit_occupancies
+      accepted_status = HmisExternalApis::AcHmis::ReferralPosting.statuses.fetch('accepted_status')
+      enrollment_pk_by_id = Hmis::Hud::Enrollment
+        .where(data_source: data_source)
+        .pluck(:enrollment_id, :id)
+        .to_h
+
+      posting_rows.each do |posting_row|
+        # only assign accepted enrollments
+        next unless posting_status(posting_row) == accepted_status
+
+        # expect enrollment_id to be populated on accepted enrollments
+        enrollment_id = row_value(posting_row, field: 'ENROLLMENT_ID')
+        enrollment_pk = enrollment_pk_by_id[enrollment_id]
+        unless enrollment_pk
+          log_skipped_row(row, field: 'ENROLLMENT_ID')
+          next
+        end
+
+        unit_type_mper_id = row_value(posting_row, field: 'UNIT_TYPE_ID')
+        unit_id = assign_next_unit(
+          enrollment_pk: enrollment_pk,
+          unit_type_mper_id: unit_type_mper_id,
+          fallback_start_date: parse_date(row_value(posting_row, field: 'STATUS_UPDATED_AT')),
+        )
```

Comment:
This will assign every member of the household to separate units.
In reality, they should all be assigned to the same unit. I didn't think about that, sorry.  I'm pushing the change!

Reply:
https://github.com/greenriver/hmis-warehouse/pull/3358/commits/b5faac9412bf9075ad2b7098297f715d2a5740b4

Reply:
Oh shoot, sorry I missed that! Thanks for fixing!